### ISSUE TITLE
Add -dtab.add flag to twitter-server.

### ISSUE THREAD

**Issue by olix0r** [twitter/twitter-server#28]
I was about to add a `dtab.add` flag to TwitterServer for my own needs, but maybe there's an implementation floating around at Twitter that you might just promote to twitter-server? ;)

Pretty please?

**Comment by olix0r** [comment]
On second thought, this may not make sense unless there's a principled way of setting the base dtab before dtab.add applies.  This could be accomplished with something like:

```scala
trait DtabApp { self: App =>
  val dtabAdd = flag("dtab.add", Dtab.empty, ...)

  def dtabInit(base: Dtab): Unit = {
    Dtab.base ++= base ++ dtabAdd()
  }
}
```

I'm willing to accept that this doesn't belong in twitter-server, though I'm curious if others think this would be generally useful.

**Comment by atollena** [comment]
Internally we use a hook to initialize the dtab:

    class DtabNewHook extends NewHook {
      def apply(app: App) = new Hook {
        val dtab = app.flag("dtab.add", Dtab.empty, "Supplemental dtab")

        def premain() {
          Dtab.base = (our base dtab) ++  dtab()

          app match {
            case l: Logging =>
              val ls = for (dentry <- Dtab.base)
                yield "\t"+dentry.show

              l.log.info("Installed Dtab:\n"+(ls mkString ";\n"))

            case _ =>
          }
        }
      }
    }

There is no way to set the base dtab to something different, before dtab.add applies. It is hardcoded here.

With what you propose we'd just need to call dtabInit("our base dtab") in this hook. That seems OK.

**Comment by cacoco** [comment]
@olix0r planning on doing this now.

**Comment by cacoco** [comment]
Merged in a00e3942d95fd0a0e29e14b47fa0997c8e6b3799

**Comment by olix0r** [comment]
🙌
